### PR TITLE
Change comments in CacheDirective to scaladoc #1514

### PR DESCRIPTION
**Pull request by tom-walford**
* Updated comments in CacheDirective to be scaladocs
* Updated comments in HttpCookie to be scaladocs
* Updated comments in LinkValue to be scaladocs

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

**Comment by jlprat** [comment]
Problem with the build is some formatting issues with the comments.
Just run sbt's `compile` task and you'll see it fixed locally. Then just push the changes

**Comment by tom-walford** [comment]
Done - really like what you guys have done with the docs - the compiler checked code is really nice

**Comment by jlprat** [comment]
Failure was https://github.com/akka/akka-http/issues/1502

PLS BUILD

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala [hunk 1]
```diff
@@ -125,8 +131,10 @@ object HttpCookie {
   import akka.http.impl.model.parser.CharacterClasses._
 
   private[http] def nameChars = tchar
-  // http://tools.ietf.org/html/rfc6265#section-4.1.1
-  // ; US-ASCII characters excluding CTLs, whitespace DQUOTE, comma, semicolon, and backslash
+  /**
+   * http://tools.ietf.org/html/rfc6265#section-4.1.1
+   * US-ASCII characters excluding CTLs, whitespace DQUOTE, comma, semicolon, and backslash
+   */
```
Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCookie.scala]: Not so important here, but thanks anyway ;)

**Comment by jrudolph** [comment]
Refs #1514.

**Pull request merged**